### DISCUSSION TITLE
module 'numpy' has no attribute 'bool'

### DISCUSSION THREAD

**Discussion by lililii124** [pypr/pysph#380] [Q&A]
On my Linux virtual machine, I used conda to create a virtual environment of pyhton3.9, and quickly installed pysph through pip install pysph, but when I ran the example, the following error was prompted：
  np.dtype(np.bool): 'char',
  File "D:\anaconda3\envs\python39\lib\site-packages\numpy\__init__.py", line 305, in __getattr__
    raise AttributeError(__former_attrs__[attr])
AttributeError: module 'numpy' has no attribute 'bool'.
`np.bool` was a deprecated alias for the builtin `bool`. To avoid this error in existing code, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations

what should i do please

**Comment by prabhuramachandran** [comment]
Sorry about this, the issue should have been fixed in the latest PySPH that is on github. You could install that using:
`pip install https://github.com/pypr/pysph/zipball/master`

At some point we need to update the version on pypi.

**Reply by lililii124** [reply]
I did as you said,but:"  File "/home/user/miniconda3/envs/pysphs/lib/python3.9/site-packages/pysph/base/utils.py", line 7, in <module>
    from .particle_array import ParticleArray, \
  File "pysph/base/particle_array.pyx", line 1, in init pysph.base.particle_array
SystemError: execution of module cyarray.carray raised unreported exception",do you know why

**Comment by prabhuramachandran** [comment]
Could you uninstall pysph first, `pip uninstall pysph` and then run the above?

**Reply by lililii124** [reply]
miniconda3/envs/pysphs/lib/python3.9/site-packages/pysph/base/tests/test_device_helper.py - SystemError: execution of module cyarray.carray raised unreported exception，yes，i pip uninstall pysph

**Comment by nauaneed** [comment]
I think this error is from compyle and it could be due to this: https://github.com/pypr/compyle/issues/97

Could you try installing compyle from Github instead of PyPI with: 
`pip install https://github.com/pypr/compyle/zipball/master`

This should solve the issue. If its still unresolved, please try reinstalling PySPH and compyle from Github after uninstalling them.
